flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java

```diff
@@ -122,20 +129,16 @@ public void cancel() {
 
     /** Implementation of a BigQuery read client wrapper. */
     public static class StorageReadClientImpl implements StorageReadClient {
-        private static final HeaderProvider USER_AGENT_HEADER_PROVIDER =
-                FixedHeaderProvider.create(
-                        "user-agent", "Apache_Flink_Java/" + FlinkVersion.current().toString());
-
         private final BigQueryReadClient client;
 
         private StorageReadClientImpl(CredentialsOptions options) throws IOException {
             BigQueryReadSettings.Builder settingsBuilder =
                     BigQueryReadSettings.newBuilder()
                             .setCredentialsProvider(
                                     FixedCredentialsProvider.create(options.getCredentials()))
+                            .setHeaderProvider(USER_AGENT_HEADER_PROVIDER)
```

Comment:
Just to clarify, this header will be provided in both CreateReadSession and ReadRows call, right?

Reply:
Yes. The same BigQueryReadClient obtained from these settings is utilized for CreateReadSession and ReadRows call.